composer/datasets/cifar.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from typing import Any, Optional
+
+import numpy as np
+from PIL import Image
+from torchvision import transforms
+
+from composer.datasets.streaming import StreamingDataset
+
+__all__ = ["StreamingCIFAR10"]
+
+
+class StreamingCIFAR10(StreamingDataset):
+    """
+    Implementation of the CIFAR10 dataset using StreamingDataset.
+
+    Args:
+        remote (str): Remote directory (S3 or local filesystem) where dataset is stored.
+        local (str): Local filesystem directory where dataset is cached during operation.
+        is_train (bool): Whether is training split.
+        shuffle (bool): Whether to shuffle the samples in this dataset.
+        batch_size (Optional[int]): Hint batch_size that will be used on each device's DataLoader. Default: ``None``.
+    """
+
+    def decode_image(self, data: bytes) -> Image.Image:
+        """Decode the sample image.
+
+        Args:
+            data (bytes): The raw bytes.
+
+        Returns:
+            Image: PIL image encoded by the bytes.
+        """
+        arr = np.frombuffer(data, np.uint8)
+        arr = arr.reshape(32, 32, 3)
+        return Image.fromarray(arr)
+
+    def decode_class(self, data: bytes) -> np.int64:
+        """Decode the sample class.
+
+        Args:
+            data (bytes): The raw bytes.
+
+        Returns:
+            np.int64: The class encoded by the bytes.
+        """
+        return np.frombuffer(data, np.int64)[0]
+
+    def __init__(self, remote: str, local: str, is_train: bool, shuffle: bool, batch_size: Optional[int] = None):
```

Comment:
Could we replace `is_train: bool` with a `split: str`? I know it's different from the original torchvision dataset but I think it's more clear.

Reply:
Done